Terminal.Gui/Views/Scroll/ScrollButton.cs

```diff
@@ -0,0 +1,142 @@
+﻿#nullable enable
+
+namespace Terminal.Gui;
+
+internal enum VariationMode
+{
+    Decrease,
+    Increase
+}
+
+internal class ScrollButton : View
+{
+    public ScrollButton ()
+    {
+        TextAlignment = Alignment.Center;
+        VerticalTextAlignment = Alignment.Center;
+        Id = "scrollButton";
+
+        //Width = Dim.Auto (DimAutoStyle.Content, 1);
+        //Height = Dim.Auto (DimAutoStyle.Content, 1);
+        WantContinuousButtonPressed = true;
+    }
+
+    private ColorScheme? _savedColorScheme;
+
+    public void AdjustButton ()
+    {
+        if (!IsInitialized)
+        {
+            return;
+        }
+
+        Width = SupView.Orientation == Orientation.Vertical ? Dim.Fill () : 1;
+        Height = SupView.Orientation == Orientation.Vertical ? 1 : Dim.Fill ();
+
+        switch (VariationMode)
+        {
+            case VariationMode.Decrease:
+                X = 0;
+                Y = 0;
+
+                break;
+            case VariationMode.Increase:
+                X = SupView.Orientation == Orientation.Vertical ? 0 : Pos.AnchorEnd (1);
+                Y = SupView.Orientation == Orientation.Vertical ? Pos.AnchorEnd (1) : 0;
+
+                break;
+            default:
+                throw new ArgumentOutOfRangeException ();
+        }
+
+        SetButtonText ();
+    }
+
+    /// <inheritdoc/>
+    public override Attribute GetNormalColor ()
+    {
+        if (_savedColorScheme is null)
+        {
+            ColorScheme = new () { Normal = new (SupView.ColorScheme.HotNormal.Foreground, SupView.ColorScheme.HotNormal.Background) };
+        }
+        else
+        {
+            ColorScheme = new () { Normal = new (SupView.ColorScheme.Normal.Background, SupView.ColorScheme.Normal.Foreground) };
+        }
+
+        return base.GetNormalColor ();
+    }
+
+    public VariationMode VariationMode { get; init; }
+
+    /// <inheritdoc/>
+    protected internal override bool? OnMouseEnter (MouseEvent mouseEvent)
+    {
+        _savedColorScheme ??= SupView.ColorScheme;
+
+        ColorScheme = new ()
+        {
+            Normal = new (_savedColorScheme.HotNormal.Foreground, _savedColorScheme.HotNormal.Foreground),
+            Focus = new (_savedColorScheme.Focus.Foreground, _savedColorScheme.Focus.Foreground),
+            HotNormal = new (_savedColorScheme.Normal.Foreground, _savedColorScheme.Normal.Foreground),
+            HotFocus = new (_savedColorScheme.HotFocus.Foreground, _savedColorScheme.HotFocus.Foreground),
+            Disabled = new (_savedColorScheme.Disabled.Foreground, _savedColorScheme.Disabled.Foreground)
+        };
+
+        return base.OnMouseEnter (mouseEvent);
+    }
+
+    /// <inheritdoc/>
+    protected internal override bool OnMouseEvent (MouseEvent mouseEvent)
```

Comment:
Seems to me that this would be more clear if it was a `MouseClicked` handler vs. override of OnMouseEvent?

Reply:
I don't think so because it's handling the `Button1Pressed` since the `WantContinuousButtonPressed` is true. I doubt the `MouseClicked` handles with this because it's needed a `Button1Released` to chain a `Button1Clicked`.